### ISSUE TITLE
multiple grep --color=always issues

### ISSUE THREAD

**Issue by hdaz000700** [koalaman/shellcheck#2895]
#### For new checks and feature suggestions
- [  ] https://www.shellcheck.net/ (i.e. the latest commit) currently gives no useful warnings about this

shellcheck does not warn or know about grep + color issues, inserting color codes in

```shell
$ cat test_file
something_1
something_2
something_BLANK
something1
something2
something3
somethingBLANK
```

BAD!
```shell
$ grep --color=always "some" test_file  | grep -v "something_BLANK"
something_1
something_2
something_BLANK
something1
something2
something3
somethingBLANK
```

BAD!
```shell
$ grep --color=always "some" test_file  | grep -v "somethingBLANK"
something_1
something_2
something_BLANK
something1
something2
something3
somethingBLANK
```

BAD!
```shell
$ grep --color=always "some" test_file  | grep --color=none -v "somethingBLANK"
something_1
something_2
something_BLANK
something1
something2
something3
somethingBLANK
```

example showing issue.

```shell
$ grep --color=always "some" test_file | tee /tmp/1
          something_1
          something_2
          something_BLANK
          something1
          something2
          something3
          somethingBLANK

$ vim /tmp/1
          ^[[01;31m^[[Ksome^[[m^[[Kthing_1
          ^[[01;31m^[[Ksome^[[m^[[Kthing_2
          ^[[01;31m^[[Ksome^[[m^[[Kthing_BLANK
          ^[[01;31m^[[Ksome^[[m^[[Kthing1
          ^[[01;31m^[[Ksome^[[m^[[Kthing2
          ^[[01;31m^[[Ksome^[[m^[[Kthing3
          ^[[01;31m^[[Ksome^[[m^[[KthingBLANK
```
Why would this be useful to highlight this behaviour to end users, it is very easy to test something on the command line and and add it to a script and not even notice "grep --color" or  "grep --color=always" is part and then get the wrong results from the script

- [X] I searched through https://github.com/koalaman/shellcheck/issues and didn't find anything related

#### Here's what shellcheck currently says:

nothing

#### Here's what I wanted or expected to see:

shellcheck to test for if grep --color is being used and warn / highlight these issues.
